### ISSUE TITLE
Use custom logger instead of root logger

### ISSUE THREAD

**Issue by slide333333** [grpc/grpc#13117]
Hi,
as far as I can see all logging is done using the root logger instead of defining a custom logger. This makes it difficult to configure the logging for larger applications.

**Comment by nathanielmanistaatgoogle** [comment]
This may just be a matter of my not being up-to-speed with current best practices about logging - can you recommend an article, guide, or some other documentation I should read on the matter? Also happy to review a pull request for this.

**Comment by nathanielmanistaatgoogle** [comment]
@slide333333: go into more detail about what the solution to this might be? Would it be satisfactory just to adopt the pattern ["A good convention to use when naming loggers is to use a module-level logger, in each module which uses logging"](https://docs.python.org/3/howto/logging.html#advanced-logging-tutorial) or do you think something more sophisticated (like per-`grpc.Channel` and per-`grpc.Server` logger instances) should be done?

**Comment by SVilgelm** [comment]
+1 for `logger = logging.getLogger(__name__)`
One of the biggest Python Projects - OpenStack uses this format.

**Comment by ghostwriternr** [comment]
Hey @nathanielmanistaatgoogle, I'd like to attempt this issue. I'm looking to do GSoC this year under grpc, so I was looking at some P2 bugs that I can try. Do let me know if you feel another issue would be more appropriate though since this would be my first time contributing to grpc. I haven't narrowed down on one of the 4 projects yet, but I have my eyes on one of the first 3 entries in [the ideas page](https://github.com/grpc/grpc/blob/master/summerofcode/ideas.md).

I just have a few questions though:
1. When I build the project and then run the tests as given in [`run_tests`](https://github.com/grpc/grpc/tree/master/tools/run_tests) (specifically, 
```sh
python tools/run_tests/run_tests.py -l python -c dbg --compiler python3.6
```
), I see a *PASSED* message as so:
```sh
PASSED: /home/ghost_000/github/grpc/tools/run_tests/helper_scripts/build_python.sh [time=803.2sec, retries=0:0]
```
But then there's a series of other errors, each saying: *failed to detect port server*. I could trace back the error to the fact that it's trying to connect to a specific `grpc-testing` project on bigquery, but of course I don't have access to it so it fails. Am I correct about this? If yes, is it safe to ignore these?

2. The contributing instructions in [`INSTALL.md`](https://github.com/grpc/grpc/blob/master/INSTALL.md) currently builds the `v1.9.x` branch. Should my PRs be to this branch or `master`?

3. Since this is a python specific issue, is there some way to build (and test) only the python related code and test it? (the complete build + running tests take close to 40 minutes on my machine).

**Comment by ghostwriternr** [comment]
cc @mehrdada @kpayson64 @hsaliak 
Tagging you since you are listed on the ideas page as potential mentors. Hope that's okay. Can someone please advice on the above?

**Comment by kpayson64** [comment]
@nathanielmanistaatgoogle or @hsaliak are probably your best points of contact for anything GSOC related.

**Comment by hsaliak** [comment]
@kpayson64  -  @ghostwriternr wants to tackle some help wanted issues in preparation for gsoc, not do this as part of gsoc itself.  

@ghostwriternr 
For (1) I think these statements can be safely ignored, but to get rid of them, can you try adding ``--disable_auto_set_flakes`` to your ``run_tests.py`` invocation? We recently added the ability to re-run tests that have been historically been flaky, so we can remove false negatives from our test runs.. disabling this check will likely remove the need for you to query this data from bigquery.   

FYI @dgquintas  -- we may want to embellish the help of this function to indicate that the failure can be ignored, and that the disable flag can be used to avoid it altogether if ``grpc-testing`` is not accessible.

2. Yes always track the master branch in your development. The core team developers do this, so to send patches, this is the path of least friction for you. If you ensure that a patch builds against master, then you make it easier for people to merge. Otherwise, reviews would have to consider this possibility as well. 
3. @nathanielmanistaatgoogle could you provide the guidance here? Perhaps running ``src/python/grpcio_tests`` after an in-place build, for quick iterations, and the full run_tests before submitting a PR?

**Comment by ghostwriternr** [comment]
Thanks a ton for the detailed reply, @hsaliak! :smile: 

I've added a very minimal PR to use the module-level logger above. It changes little, but thanks to your comment, I can see how the code is structured now. The `disable_auto_set_flakes` was indeed helpful and resolved the bigquery issue.

The [`build_python.sh`](https://github.com/grpc/grpc/blob/master/tools/run_tests/helper_scripts/build_python.sh) and [`run_python.sh`](https://github.com/grpc/grpc/blob/master/tools/run_tests/helper_scripts/run_python.sh) scripts contained the steps to follow for running the tests individually. It was indeed using `src/python/grpcio_tests/setup.py test_lite` to run the tests.

**Comment by nathanielmanistaatgoogle** [comment]
@slide333333, @SVilgelm, @davidt99: any of you want to review [pull request 14557](https://github.com/grpc/grpc/pull/14557) which is intended to fulfill this issue?

**Comment by davidt99** [comment]
@nathanielmanistaatgoogle, reviewed.

**Comment by nathanielmanistaatgoogle** [comment]
This has been fixed by @ghostwriternr and the fix will be released in a few weeks in gRPC Python 1.13.
